src/lib.rs

```diff
@@ -111,7 +115,7 @@ pub fn convert_fee_rate(target: usize, estimates: HashMap<u16, f64>) -> Option<f
 }
 
 #[derive(Debug, Clone)]
-pub struct Builder {
+pub struct Builder<S = DefaultSleeper> {
```

Comment:
Why not move `DefaultSleeper` to lib.rs so that it's visible to all features? Granted, `DefaultSleeper` is only a `Sleeper` implementation when tokio is enabled, so it's basically meaningless in any other context. I don't really like that the generic bleeds over into the Builder struct but I'm not finding a better way to do it either.

Reply:
ya that works, done

Reply:
Troubleshooting the CI fail, I changed every instance of `#[cfg(all(feature = "blocking", feature = "async"))]` in the unit tests to use "tokio" instead of "async", but there's probably a better way to organize the test feature configs.

Reply:
Fixed the test, forgot to add `async-trait` to it

Reply:
I don't think this would be a good change, currently most end users will not have to deal with the traits at all. Anyone using `DefaultSleeper` will not need any code changes.

@ValuedMammal Tried an implementation with `dyn Sleeper` and was not able to keep the clone impl on the async client.

The generics make the code a bit more complex for the tradeoff of keeping things simple for the user. I think thats a worthwhile tradeoff.

Reply:
As for the `async_trait` overhead, I think that mostly just boxes the future too. We can completely avoid that overhead when MSRV is increased. Or we can add some complexity now and only require `async_trait` if you're not using a version of rust that has the async_trait feature.

Reply:
@tnull This was the alternate approach following the idea of passing a sleep closure https://github.com/ValuedMammal/rust-esplora-client/commit/d83ea69bdb867b8f7ae313294f91751d3dd265ca. Not sure if there's a way to do this while keeping `Clone` for `AsyncClient`

Reply:
I'm not really in favor of adding yet another dependency, in this scenario the `async_trait` one, unless it's the only option.

I'd choose the closure approach over any other, as it seems way simpler and doesn't disturb the API, but, @ValuedMammal already has tried, and I also gave it a try, but I had no success with the `impl Clone` problem.

However, I don't think the `async_trait` is mandatory here, I took a look at other crates and gave it a try, and it seems to work without it at #107.

Let me know what you guys think cc @praveenperera @ValuedMammal.

Reply:
I would be in favor of dropping `async-trait` by adding the associated type `type Sleep: Future<Output = ()>` to the `Sleeper` trait